include/deal.II/base/discrete_time.h

```diff
@@ -0,0 +1,132 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2019 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_discrete_time_h
+#define dealii_discrete_time_h
+
+#include <deal.II/base/exceptions.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * Provides a means to keep track of the simulation time in a time-dependent
+ * simulation. It manages stepping forward from an initial time to a final
+ * time. It only allows fixed time step size, although the step size may be
+ * automatically adjusted slightly to capture the final time.
+ *
+ * @author Reza Rastak, 2019
+ */
+class DiscreteTime
+{
+public:
+  /**
+   * Constructor
+   */
+  DiscreteTime(const double start_time,
+               const double end_time,
+               const double start_step_size);
+
+  /**
+   * Return the current time.
+   */
+  double
+  get_current_time() const;
+
+  /**
+   * Return the ends of the time interval.
+   * The final time step ends exactly at this point.
+   */
+  double
+  get_end_time() const;
+
+  /**
+   * Return the size of the step from current time step to the next.
+   */
+  double
+  get_next_step_size() const;
+
+  /**
+   * Set the value of the next time step size.
+   */
+  void
+  set_next_step_size(const double time_step_size);
+
+  /**
+   * Advance the current time based on the value of the current step.
+   * Afterward, it modifies the value of the time step size if needed.
+   */
+  void
+  advance_time();
```

Comment:
Can you say in which circumstances it modifies the time step size?

Reply:
I improved the documentation here.